src/main.rs

```diff
@@ -1,3 +1,76 @@
+mod memory;
+mod opdcodes;
+mod registers;
+
+use memory::Memory;
+use opdcodes::Opcode;
+use registers::Registers;
+
+struct VM {
+    memory: Memory,
+    registers: Registers,
+}
+
+impl VM {
+    pub fn new() -> Self {
+        Self {
+            memory: Memory::new(),
+            registers: Registers::new(),
+        }
+    }
+
+    pub fn read_memory(&self, address: u16) -> u16 {
+        self.memory.read(address)
+    }
+
+    pub fn write_memory(&mut self, address: u16, value: u16) {
+        self.memory.write(address, value);
+    }
+
+    pub fn read_register(&self, r: usize) -> u16 {
+        self.registers.get(r).unwrap_or(0)
+    }
+
+    pub fn write_register(&mut self, r: usize, value: u16) {
+        self.registers.set(r, value);
+    }
+
+    pub fn update_flags(&mut self, r: usize) {
+        self.registers.update_flags(r);
+    }
+
+    pub fn run(&mut self) {
+        loop {
+            // 1. Load one instruction from memory at the address of the PC
+            let instruction = self.read_memory(self.registers.pc);
+
+            // 2. Increment the PC
+            self.registers.pc = self.registers.pc.wrapping_add(1);
+
+            let opcode: Opcode = Opcode::from((instruction >> 12) & 0xF);
```

Comment:
Slight suggestion, what do you think about storing the bit shifted instruction inside a variable? 

This is simply my personal opinion, but I think it makes debugging easier. I know a special someone that would aGRee, yoU probably know him, riGht?

Something like: 
```suggestion
            let instruction = (instruction >> 12) & 0xF; 
            let opcode: Opcode = Opcode::from(instruction);
```
Maybe use a different variable name if you prefer